src/Hyperion/Benchmark.hs

```diff
@@ -55,7 +60,37 @@ instance Show Benchmark where
   showsPrec x (WithSampling _opt bk) = showsPrec x bk
 
 bench :: String -> Batch () -> Benchmark
-bench name batch = Bench (Text.pack name) batch
+bench name batch = Bench Text.pack (const []) id (const batch) name
+
+benchWith
+  :: (a -> Text)
+  -> (a -> [JSON.Pair])
+  -> (a -> b)
+  -> (b -> Batch ())
+  -> a -> Benchmark
+benchWith = Bench
+
+benchMany
```

Comment:
In short, I understand the problem, but I'm not convinced that this is the right solution space for this problem.